enterprise/coderd/workspaceproxy.go

```diff
@@ -127,23 +127,85 @@ func (api *API) patchWorkspaceProxy(rw http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	updatedProxy, err := api.Database.UpdateWorkspaceProxy(ctx, database.UpdateWorkspaceProxyParams{
-		Name:        req.Name,
-		DisplayName: req.DisplayName,
-		Icon:        req.Icon,
-		ID:          proxy.ID,
-		// If hashedSecret is nil or empty, this will not update the secret.
-		TokenHashedSecret: hashedSecret,
-	})
-	if httpapi.Is404Error(err) {
-		httpapi.ResourceNotFound(rw)
-		return
-	}
+	deploymentIDStr, err := api.Database.GetDeploymentID(ctx)
 	if err != nil {
 		httpapi.InternalServerError(rw, err)
 		return
 	}
 
+	var updatedProxy database.WorkspaceProxy
+	if proxy.ID.String() == deploymentIDStr {
```

Comment:
Since the code for this route is completely different for primary and other proxies, could you register it as a different route on the handler? like `PUT /api/v2/proxies/primary`, `PUT /api/v2/proxies/${deployment_id}`, and `PUT /api/v2/proxies/:other_id`.

Reply:
I factored it out into it's own function. It's own route got a bit tricky since I had to duplicate route blocks.